bson/__init__.py

```diff
@@ -1025,9 +1025,21 @@ def encode(
     return _dict_to_bson(document, check_keys, codec_options)
 
 
+@overload
+def decode(data: "_ReadableBuffer", codec_options: None = None) -> Dict[str, Any]:
```

Comment:
Nice! This is great! I think we want to apply this same overload pattern throughout the codebase but I think that's too much for this PR. Could you open a new ticket for it? Eg decode_all, decode_iter, and anywhere else we use _DocumentType.

Reply:
sounds good, made the ticket